includes/cli/class-data-backfill.php

```diff
@@ -0,0 +1,362 @@
+<?php
+/**
+ * Data Backfill scripts.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network;
+
+use Automattic\Jetpack\VideoPress\Data;
+use WP_CLI;
+
+/**
+ * Data Backfill class.
+ */
+class Data_Backfill {
+	/**
+	 * The final results object.
+	 *
+	 * @var array
+	 */
+	private static $results = [
+		'reader_registered' => [
+			'processed' => 0,
+			'duplicate' => 0,
+		],
+		'donation_new'      => [
+			'processed' => 0,
+			'duplicate' => 0,
+		],
+	];
+
+	/**
+	 * WP_CLI progress handler.
+	 *
+	 * @var class
+	 */
+	private static $progress = false;
+
+	/**
+	 * Initialize this class and register hooks
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		add_action( 'init', [ __CLASS__, 'register_commands' ] );
+	}
+
+	/**
+	 * Register the WP-CLI commands
+	 *
+	 * @return void
+	 */
+	public static function register_commands() {
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::add_command( 'newspack-network data-backfill', [ __CLASS__, 'data_backfill' ] );
+		}
+	}
+
+	/**
+	 * Find existing webhook requests for a given action and data.
+	 *
+	 * @param string $action The action name.
+	 * @param int    $timestamp The timestamp.
+	 * @param array  $data The data.
+	 */
+	private static function find_webhook_requests( $action, $timestamp, $data ) {
+		return get_posts(
+			[
+				'post_type'   => \Newspack\Data_Events\Webhooks::REQUEST_POST_TYPE,
+				'post_title'  => $action,
+				'post_status' => 'any',
+				'meta_query'  => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+					'relation' => 'AND',
+					[
+						'key'     => 'timestamp',
+						'value'   => $timestamp,
+						'compare' => '=',
+					],
+					[
+						'key'     => 'action_name',
+						'value'   => $action,
+						'compare' => '=',
+					],
+					[
+						'key'     => 'data',
+						'value'   => wp_json_encode( $data ),
+						'compare' => '=',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Process reader_registered events.
+	 *
+	 * @param string $start The start date.
+	 * @param string $end The end date.
+	 * @param bool   $live Whether to run in live mode.
+	 * @param bool   $verbose Whether to output verbose information.
+	 */
+	private static function process_reader_registered( $start, $end, $live, $verbose ) {
+		// Get all users registered between $start and $end.
+		$users = get_users(
```

Comment:
We could filter out users with `newspack_remote_id` metadata. No need to backfill users that came from other sites. We want users that were originally registered on this site

Reply:
Oh, great suggestion! Done in 111a7b4fde706d7f6de42f2237ee619e8a36b477